dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -680,8 +683,8 @@ Spdp::handle_participant_data(DCPS::MessageId id,
     // Since we've just seen a new participant, let's send out our
     // own announcement, so they don't have to wait.
     if (from != ACE_INET_Addr()) {
-      if (from_relay) {
-        tport_->write_i(guid, SpdpTransport::SEND_TO_RELAY);
+      if (!from_relay && from_relay) {
```

Comment:
Wouldn't this always be false?

Reply:
Good catch.  That probably saved me hours of debugging.